controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,548 @@
+						Eventually(func() error {
+							err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+								Namespace: namespace2.GetName(),
+							},
+								client.MatchingLabels{
+									scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+									scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+									clusterRoleBindingGenerateKey: "test",
+								})
+							if err != nil {
+								return err
+							}
+
+							if len(roleBindingList.Items) != 1 {
+								return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+							}
+
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						existingRB := &roleBindingList.Items[0]
+
+						Expect(len(existingRB.OwnerReferences)).To(Equal(1))
+						Expect(existingRB.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+							APIVersion: "operators.io.operator-framework/v1",
+							Kind:       "ScopeInstance",
+							Name:       scopeInstance.GetObjectMeta().GetName(),
+							UID:        scopeInstance.GetObjectMeta().GetUID(),
+						}))
+
+						Expect(len(existingRB.Subjects)).To(Equal(1))
+						Expect(existingRB.Subjects).Should(ContainElement(rbacv1.Subject{
+							Kind:     "Group",
+							Name:     "manager",
+							APIGroup: "rbac.authorization.k8s.io",
+						}))
+						Expect(existingRB.RoleRef).To(Equal(rbacv1.RoleRef{
+							Kind:     "ClusterRole",
+							Name:     "test",
+							APIGroup: "rbac.authorization.k8s.io",
+						}))
+
+						roleBindingList = &rbacv1.RoleBindingList{}
+						Eventually(func() error {
+							err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+								Namespace: namespace.GetName(),
+							},
+								client.MatchingLabels{
+									scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+									scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+									clusterRoleBindingGenerateKey: "test",
+								})
+							if err != nil {
+								return err
+							}
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						Expect(len(roleBindingList.Items)).To(Equal(0))
+					})
+
+					When("a scopeInstance is updated to remove all namespaces", func() {
+						It("should not create a clusterRole", func() {
+							Eventually(func() error {
+								// TODO: scopeTemplate should check for status that no clusterRole exists.
+								// Until then, just make sure no clusterRole exists
+								clusterRoleList := &rbacv1.ClusterRoleList{}
+								if err := k8sClient.List(ctx, clusterRoleList, client.MatchingLabels{
+									scopeTemplateUIDKey:    string(scopeTemplate.GetUID()),
+									clusterRoleGenerateKey: "test",
+								}); err != nil {
+									return err
+								}
+
+								if len(clusterRoleList.Items) != 0 {
+									return fmt.Errorf("Expected 0 clusterRoles, found %d", len(clusterRoleList.Items))
+								}
+
+								return nil
+
+							}, timeout, interval).Should(BeNil())
+						})
```

Comment:
This seems wrong give the when above which says the scopeInstance removed all namespaces.

Reply:
this is to verify the ClusterRole. Even if you remove all namespaces, the clusterrole'll be there. Correct me if I am wrong.

Reply:
agree. updated